src/lib/entities/entry.ts

```diff
@@ -1,17 +1,20 @@
 import { cloneDeep } from 'lodash'
 import APIEntry from '../interfaces/api-entry'
+import { isNullOrUndefined } from 'util'
 
 class Entry {
   private _id: string
   private _contentTypeId: string
   private _version: number
   private _fields: object
+  private _isPublished: boolean
 
   constructor (entry: APIEntry) {
     this._id = entry.sys.id
     this._fields = entry.fields
     this._version = entry.sys.version
     this._contentTypeId = entry.sys.contentType.sys.id
+    this._isPublished = !isNullOrUndefined(entry.sys.publishedVersion)
```

Comment:
When an entry is cloned, we call the constructor with `this.toApiEntry`, but that does not include the `publishedVersion`. I think that may cause issues, so probably we should also include the `publishedVersion` in the clone.

We should keep in mind to remove `publishedVersion` then after an `unpublish` though, so that the next clone does not restore the published state.